exchange/utils.go

```diff
@@ -819,37 +809,15 @@ func removeUnpermissionedEids(request *openrtb2.BidRequest, bidder string, reque
 		return nil
 	}
 
-	// marshal eidsAllowed back to userExt
+	// set eidsAllowed back to userExt
 	if len(eidsAllowed) == 0 {
-		delete(userExt, "eids")
+		reqWrapper.User.EIDs = nil
 	} else {
-		eidsRaw, err := jsonutil.Marshal(eidsAllowed)
-		if err != nil {
-			return err
-		}
-		userExt["eids"] = eidsRaw
-	}
-
-	// exit early if userExt is empty
-	if len(userExt) == 0 {
-		setUserExtWithCopy(request, nil)
-		return nil
+		reqWrapper.User.EIDs = eidsAllowed
 	}
```

Comment:
Do we need both cases anymore? If `eidsAllowed` is null when there are no eids allowed, then we can get rid of the if statement and just set `reqWrapper.User.EIDs = eidsAllowed`.

Reply:
I think we can remove the conditional here if we change
`eidsAllowed := make([]openrtb2.EID, 0, len(eids))` to `var eidsAllowed []openrtb2.EID`
If we do that `eidsAllowed` will be start off as `nil`, however there will be the performance penalty when building up that slice via `append`.
@SyntaxNode thoughts?

Reply:
Discussed offline. We'll keep it the way it is using `make` to initialize `eidsAllowed` and the conditional check here.